From eb628eb18184703cb4abe526a1e79ba6ba2f43ab Mon Sep 17 00:00:00 2001
From: G Davey <gdavey@committhis.co.uk>
Date: Tue, 28 Jul 2020 12:00:33 +0100
Subject: [PATCH 1/1] Add smbus header to onlp i2c source

This replaces the custom header provided by ONL. The file previously
provided was <linux/i2c-devices.h>; the ONL build system copies this
from the host it was built on and into the rootfs. To move away from
host dependent builds, and make sure that the headers are set up for
building specifically for the Mion kernel, we replace this header
with the correct definitions. The default value for all builds is to
use the custom header via ONLPLIB_CONFIG_I2C_USE_CUSTOM_HEADER, so
this provides a convenient place to add these changes without having
platform vendors to worry about it.

Signed-off-by: G Davey <gdavey@committhis.co.uk>
---
 packages/base/any/onlp/src/onlplib/module/src/i2c.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/packages/base/any/onlp/src/onlplib/module/src/i2c.c b/packages/base/any/onlp/src/onlplib/module/src/i2c.c
index f5d06955..f003413f 100644
--- a/packages/base/any/onlp/src/onlplib/module/src/i2c.c
+++ b/packages/base/any/onlp/src/onlplib/module/src/i2c.c
@@ -31,7 +31,8 @@
 #include <unistd.h>
 
 #if ONLPLIB_CONFIG_I2C_USE_CUSTOM_HEADER == 1
-#include <linux/i2c-devices.h>
+#include <linux/i2c-dev.h>
+#include <i2c/smbus.h>
 #else
 #include <linux/i2c-dev.h>
 #endif
-- 
2.17.1
